### PR TITLE
blog post detailing EMS migration to self-hosting

### PR DESCRIPTION
**Pull request by joshsimmons**
Scheduled for publication on Thursday, January 4

### CODE REVIEW

File: content/blog/2024/01/2024-01-04-migrating-from-ems-to-selfhosted-matrix.md [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+
+I deployed my new Matrix homeserver on a VPS running Ubuntu 22.04, which I was already using to host some other things. All my local work was done by command line on my MacBook. This meant I already had a server with most of the dependencies already in place, and I could readily SSH into it. It also means that I’ve already got Nginx listening on ports 80 and 443, so I had to get a little creative and dig deeper into the docs to set up [Nginx as a reverse proxy in front of Traefik](https://github.com/spantaleev/matrix-docker-ansible-deploy/blob/master/examples/nginx/README.md).
+
+A few more important facts: EMS runs the Synapse homeserver alongside the matrix-media-repo (MMR) projec, I’m doing the same to make this migration possible. And EMS, when I did my export, ran Synapse v1.97 whereas the latest release is v1.98. I confirmed there were no database schema changes between those two versions, so the difference in version number was immaterial, but typically your migration target should be the same version as your export.
```
Comment (MTRNord) [inline content/blog/2024/01/2024-01-04-migrating-from-ems-to-selfhosted-matrix.md:39]: Like the CI suggests :)

```suggestion
A few more important facts: EMS runs the Synapse homeserver alongside the matrix-media-repo (MMR) project, I’m doing the same to make this migration possible. And EMS, when I did my export, ran Synapse v1.97 whereas the latest release is v1.98. I confirmed there were no database schema changes between those two versions, so the difference in version number was immaterial, but typically your migration target should be the same version as your export.
```

File: content/blog/2024/01/2024-01-04-migrating-from-ems-to-selfhosted-matrix.md [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+* You’ve got a directory setup on the host server at `/root/migration` – and you run commands on the host as root.
+
+OK, with all that scene setting done, let’s walk through the steps you’ll need to take to set up your homeserver and import your data!
+
+### Run your Ansible playbook
+
+From your local machine, run the Ansible playbook. All of the commands we run from the local machine are within the base directory of the MDAD project. Critically, we’re only installing things – not starting any services yet.
+
+```
+> just roles
```
Comment (MTRNord) [inline content/blog/2024/01/2024-01-04-migrating-from-ems-to-selfhosted-matrix.md:83]: I might be out of date with the playbook/ansible, but is this a command and if so where is it from? :)

Comment (MTRNord) [reply content/blog/2024/01/2024-01-04-migrating-from-ems-to-selfhosted-matrix.md:83]: Specifically the `just` command

Comment (joshsimmons) [reply content/blog/2024/01/2024-01-04-migrating-from-ems-to-selfhosted-matrix.md:83]: Got that from p2: https://github.com/spantaleev/matrix-docker-ansible-deploy/blob/master/docs/installing.md

Thank you for the detailed review, the more eyes on this the better I feel about it! 😅

Comment (MTRNord) [reply content/blog/2024/01/2024-01-04-migrating-from-ems-to-selfhosted-matrix.md:83]: Ah ok. Maybe it makes sense to just link/introduce it once here in the post to keep it flowing :) The ansible commands are clear considering its an ansible playbook. :(

**Review by MTRNord — APPROVED** [review]
Apart from that comment above, it sounds sane to me. I guess @thibaultamartin wants a read as well before it's merged? Just to have another look at it? :D (see also [matrix.org room](https://matrix.to/#/#matrix.org-website:matrix.org where some more discussion happened)

**Pull request merged**